tests/tools.vitruv.applications.pcmjava.tests/src/tools/vitruv/applications/pcmjava/tests/pcm2java/Pcm2JavaTransformationTest.xtend

```diff
@@ -0,0 +1,205 @@
+			assertTrue(equalityHelper.equals(actualPackage, it), modelComparisonErrorMessage(it, actualPackage))
+		]
+	}
+
+	private def String modelComparisonErrorMessage(EObject expected, EObject actual) {
+		var printTarget = new DefaultPrintTarget()
+		var printIdProvider = new DefaultPrintIdProvider()
+		var modelPrinter = new DefaultModelPrinter()
+
+		printTarget.print("Expected actual model to equal expected model.")
+		printTarget.newLine
+		printTarget.print("Actual:")
+		printTarget.newLine
+		modelPrinter.printObject(printTarget, printIdProvider, actual)
+
+		printTarget.newLine
+		printTarget.print("Expected:")
+		printTarget.newLine
+		modelPrinter.printObject(printTarget, printIdProvider, expected)
+
+		return printTarget.toString
+	}
+
+	private def void ignoreComparisonOfJavaUtilClassifieres(EqualityHelper equalityHelper,
+		Collection<CompilationUnit> actualCompilationUnits, Collection<CompilationUnit> expectedCompilationUnits) {
+		val expectedJavaUtilCompilationUnits = expectedCompilationUnits.filter [ cu |
+			cu.namespaces.equals(#["java", "util"])
+		]
+		val actualJavaUtilCompilationUnits = actualCompilationUnits.filter[cu|cu.namespaces.equals(#["java", "util"])]
+
+		expectedJavaUtilCompilationUnits.forEach [ expectedCompilationUnit |
+			val actualCompilationUnit = actualJavaUtilCompilationUnits.filter [ acu |
+				acu.name.equals(expectedCompilationUnit.name)
+			].claimOne
+
+			val expectedClassifier = expectedCompilationUnit.classifiers.claimOne
+			val actualClassifier = actualCompilationUnit.classifiers.claimOne
+			equalityHelper.put(expectedClassifier, actualClassifier)
+			equalityHelper.put(actualClassifier, expectedClassifier)
+		]
+	}
+
+	private def void adaptClassMemberOrderOfExpectedToActualOrder(CompilationUnit expected, CompilationUnit actual) {
+		expected.classifiers.forEach [ expectedClassifier, classifierIdx |
+			val actualClassifier = actual.classifiers.get(classifierIdx)
+
+			ECollections.sort(expectedClassifier.members, Comparator.comparing [ item |
+				var maybeIndexInActualOfItem = IntStream.range(0, actualClassifier.members.size()).filter([ idx |
+					val actualMemberAtIdx = actualClassifier.members.get(idx)
+
+					var namesEqualOrNotPresent = true
+					if (actualMemberAtIdx instanceof NamedElement && item instanceof NamedElement) {
+						namesEqualOrNotPresent = (actualMemberAtIdx as NamedElement).name.equals(
+							(item as NamedElement).name)
+					}
+
+					return actualMemberAtIdx.class == item.class && namesEqualOrNotPresent
+				]).findFirst
+				maybeIndexInActualOfItem.present ? maybeIndexInActualOfItem.asInt : actualClassifier.members.size
+			])
+		]
+	}
+
+	// === Misc ===
+	private def <T> boolean hasDuplicate(Iterable<T> all) {
+		val Set<T> set = new HashSet<T>();
+		// Set#add returns false if the set does not change, which
+		// indicates that a duplicate element has been added.
+		for (T each : all)
+			if(!set.add(each)) return true;
+		return false;
+	}
```

Comment:
```suggestion
	private def <T> boolean hasDuplicate(Collection<T> all) {
		val Set<T> set = new HashSet<T>(all);
		return set.size() != all.size();
	}
```

I think this would work too right? Probably a bit more self-explaining

Reply:
Works too. But changes the signature of the method ;)

Reply:
Don't see a problem with that, as it is a private method which is only used with lists.

Reply:
I prefer the more compact version Jan proposed! 👍

Reply:
@h4uges after this comment is resolved, the PR is ready to be merged